chain/syncer.go

```diff
@@ -324,14 +291,34 @@ func (syncer *Syncer) widen(ctx context.Context, ts types.TipSet) (types.TipSet,
 	return wts, nil
 }
 
+// NewPeerHeadInfo accepts a peer, its current head state and its trustworthiness
+// and returns a PeerHeadInfo.
+func NewPeerHeadInfo(head types.TipSetKey, height types.Uint64, from peer.ID, trusted bool) *PeerHeadInfo {
```

Comment:
This Method and its corresponding type need a better home, looking for suggestions, else will leave it here.

Reply:
On second thought, going to wait for #3163 to land and rebase. `types.ChainInfo` will then replace this.

Reply:
As I noted there, I expect the `Trusted` field to be removed from that structure, but that it's perfectly fine for that to come as an explicit argument to `HandleNewTipSet`.